packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,218 @@
+    await this.executeCommand(
+      this.buildRetrieveOrgSourceCommand(
+        data.usernameOrAlias,
+        tempMetadataPath,
+        manifestPath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3: unzip retrieved source
+    const unpackagedZipFile = path.join(tempMetadataPath, 'unpackaged.zip');
+    try {
+      const zip = new AdmZip(unpackagedZipFile);
+      zip.extractAllTo(tempMetadataPath, true);
+    } catch (error) {
+      this.reportError('error_extracting_org_source', error);
+      return Promise.reject();
+    }
+
+    // 4: convert org source to decomposed (source) format
+    const unconvertedSourcePath = path.join(tempMetadataPath, 'unpackaged');
+    const convertedSourcePath = path.join(tempMetadataPath, 'converted');
+    await this.executeCommand(
+      this.buildMetadataApiConvertOrgSourceCommand(
+        unconvertedSourcePath,
+        convertedSourcePath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 5: diff project directory (local) and retrieved directory (remote)
+    // Assume there are consistent subdirs from each root i.e. 'main/default'
+    const localSourcePath: string = path.join(projectPath, data.packageDir);
+    const diffs = this.differ.diff(localSourcePath, convertedSourcePath);
+
+    // 6: cleanup temp directory
+    try {
+      shell.rm('-rf', tempMetadataPath);
+    } catch (error) {
+      // Failed cleanup should not terminate the service
+      this.reportError('error_cleanup_temp_files', error);
+    }
+
+    return diffs;
+  }
+
+  public async executeCommand(
+    command: Command,
+    projectPath: string,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ): Promise<string> {
+    const startTime = process.hrtime();
+    const execution = new CliCommandExecutor(
+      command,
+      { cwd: projectPath, env: { SFDX_JSON_TO_STDOUT: 'true' } },
+      true
+    ).execute(cancellationToken);
+
+    const result = new CommandOutput().getCmdResult(execution);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    execution.processExitSubject.subscribe(() => {
+      telemetryService.sendCommandEvent(execution.command.logName, startTime);
+    });
+    return result;
+  }
+
+  protected attachExecution(
+    execution: CommandExecution,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ) {
+    channelService.streamCommandOutput(execution);
+    channelService.showChannelOutput();
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private reportError(messageKey: string, error: Error) {
+    console.error(error);
```

Comment:
are we showing the error on the console here as well as the vs code notification?

Reply:
@AnanyaJha Yes, it shows up in both places. I'll add a test case to verify.